src/ucs/sys/sock.c

```diff
@@ -354,31 +361,52 @@ const char* ucs_sockaddr_str(const struct sockaddr *sock_addr,
     return str;
 }
 
-int ucs_sockaddr_is_equal(const struct sockaddr *sa1,
-                          const struct sockaddr *sa2,
-                          ucs_status_t *status_p)
+int ucs_sockaddr_cmp(const struct sockaddr *sa1,
+                     const struct sockaddr *sa2,
+                     ucs_status_t *status_p)
 {
+    int result          = 1;
     ucs_status_t status = UCS_OK;
-    int result          = 0;
+    uint16_t port1, port2;
 
-    if (sa1->sa_family != sa2->sa_family) {
+    if (!ucs_sockaddr_is_known_af(sa1) ||
+        !ucs_sockaddr_is_known_af(sa2)) {
+        ucs_error("unknown address family: %d",
+                  !ucs_sockaddr_is_known_af(sa1) ?
+                  sa1->sa_family : sa2->sa_family);
+        status = UCS_ERR_INVALID_PARAM;
+        goto out;
+    }
+
+    if (sa1->sa_family < sa2->sa_family) {
+        result = (sa1->sa_family < sa2->sa_family) ? -1 : 1;
         goto out;
     }
 
     switch (sa1->sa_family) {
     case AF_INET:
-        result = ((UCS_SOCKET_INET_PORT(sa1) == UCS_SOCKET_INET_PORT(sa2)) &&
-                  (memcmp(&UCS_SOCKET_INET_ADDR(sa1), &UCS_SOCKET_INET_ADDR(sa2),
-                          sizeof(UCS_SOCKET_INET_ADDR(sa1))) == 0));
+        result = ucs_signum(memcmp(&UCS_SOCKET_INET_ADDR(sa1),
+                                   &UCS_SOCKET_INET_ADDR(sa2),
+                                   sizeof(UCS_SOCKET_INET_ADDR(sa1))));
+
+        port1 = ntohs(UCS_SOCKET_INET_PORT(sa1));
+        port2 = ntohs(UCS_SOCKET_INET_PORT(sa2));
+
+        if (!result && (port1 != port2)) {
+            result = (port1 < port2) ? -1 : 1;
+        }
         break;
     case AF_INET6:
-        result = ((UCS_SOCKET_INET6_PORT(sa1) == UCS_SOCKET_INET6_PORT(sa2)) &&
-                  (memcmp(&UCS_SOCKET_INET6_ADDR(sa1), &UCS_SOCKET_INET6_ADDR(sa2),
-                          sizeof(UCS_SOCKET_INET6_ADDR(sa1))) == 0));
-        break;
-    default:
-        ucs_error("unknown address family: %d", sa1->sa_family);
-        status = UCS_ERR_INVALID_PARAM;
+        result = ucs_signum(memcmp(&UCS_SOCKET_INET6_ADDR(sa1),
+                                   &UCS_SOCKET_INET6_ADDR(sa2),
+                                   sizeof(UCS_SOCKET_INET6_ADDR(sa1))));
+
+        port1 = ntohs(UCS_SOCKET_INET6_PORT(sa1));
+        port2 = ntohs(UCS_SOCKET_INET6_PORT(sa2));
+
+        if (!result && (port1 != port2)) {
```

Comment:
this code is the same in both cases, can be moved outside the switch

Reply:
done